beetmoverscript/script.py

```diff
@@ -18,54 +18,92 @@
+    mapping_manifest = generate_beetmover_manifest(context)
+
+    # some files to-be-determined via script configs need to have their
+    # contents pretty named, so doing it here before even beetmoving begins
+    blobs = context.config.get('blobs_needing_prettynaming_contents', [])
+    alter_unpretty_contents(context, blobs, mapping_manifest)
+
+    # balrog_manifest is written and uploaded as an artifact which is used by
+    # a subsequent balrogworker task in the release graph. Balrogworker uses
+    # this manifest to submit release blob info (e.g. mar filename, size, etc)
     context.balrog_manifest = list()
 
-    # the checksums manifest is written and uploaded as an artifact which is used
-    # by a subsequent signing task and again by another beetmover task to
-    # upload it along with the other artifacts
+    # the checksums manifest is written and uploaded as an artifact which is
+    # used by a subsequent signing task and again by another beetmover task to
+    # upload it to S3
     context.checksums = dict()
 
-    # determine and validate the task schema along with its scopes
-    context.task = get_task(context.config)  # e.g. $cfg['work_dir']/task.json
-    validate_task_schema(context)
+    # for each artifact in manifest
+    #   a. map each upstream artifact to pretty name release bucket format
+    #   b. upload to corresponding S3 location
+    await move_nightly_beets(context, context.artifacts_to_beetmove,
+                             mapping_manifest)
+
+    #  write balrog_manifest to a file and add it to list of artifacts
+    add_balrog_manifest_to_artifacts(context)
+    # determine the correct checksum filename and generate it, adding it to
+    # the list of artifacts afterwards
+    add_checksums_to_artifacts(context)
+    # add release props file to later be used by beetmover jobs than upload
+    # the checksums file
+    add_release_props_to_artifacts(context, release_props_file)
 
+
+async def push_to_candidates(context):
     # determine artifacts to beetmove
     context.artifacts_to_beetmove = get_upstream_artifacts(context)
-    # determine the release properties and make a copy in the artifacts
-    # directory
+
+    # find release properties and make a copy in the artifacts directory
     release_props_file = get_initial_release_props_file(context)
     context.release_props = get_release_props(release_props_file)
 
     # generate beetmover mapping manifest
-    mapping_manifest = generate_beetmover_manifest(context.config,
-                                                   context.task,
-                                                   context.release_props)
-    # validate scopes to prevent beetmoving in the wrong place
-    validate_task_scopes(context, mapping_manifest)
+    mapping_manifest = generate_beetmover_manifest(context)
+
+    # perform another validation check against the bucket path
+    validate_bucket_paths(context.bucket, mapping_manifest['s3_bucket_path'])
```

Comment:
@escapewindow 
1. Until we have a green-go that permissions from `release creds` to push to nightly paths are removed, I say we keep this additional paths-check. Certainly couldn't hurt.
2. The reason this check is missing from `push_to_nightly` is that I haven't yet touched the nightly templates to reflect the *new standard* (of using a separate pretty naming + destinations in the templates). Once we're past the alpha/beeta tesing I'll make the modifications and adapt the nightly templates too so we can merge the two `move_*_beets` functions.